integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchemaCommon.scala

```diff
@@ -473,6 +528,11 @@ class AlterTableColumnSchemaGenerator(
       for (elem <- alterTableModel.tableProperties) {
         if (elem._1.toLowerCase.startsWith(defaultValueString)) {
           if (col.getColumnName.equalsIgnoreCase(elem._1.substring(defaultValueString.length))) {
+            if (DataTypes.isArrayType(col.getDataType) || DataTypes.isStructType(col.getDataType) ||
+                DataTypes.isStructType(col.getDataType)) {
```

Comment:
Same check twice for Struct type. Please check

Reply:
done